pkg/segment/query/querystatus.go

```diff
@@ -925,3 +937,43 @@ func LogGlobalSearchErrors(qid uint64) {
 		putils.LogUsingLevel(errInfo.LogLevel, "qid=%v, %v, Count: %v, ExtraInfo: %v", qid, errMsg, errInfo.Count, errInfo.Error)
 	}
 }
+
+func SetNewQueryPipelineResponse(response *structs.PipeSearchResponseOuter, qid uint64) error {
+	arqMapLock.RLock()
+	rQuery, ok := allRunningQueries[qid]
+	arqMapLock.RUnlock()
+	if !ok {
+		log.Errorf("SetQueryResponse: qid %+v does not exist!", qid)
+		return fmt.Errorf("qid does not exist")
+	}
+
+	rQuery.rqsLock.Lock()
+	defer rQuery.rqsLock.Unlock()
+	rQuery.NewPipeLineResponse = response
+	rQuery.totalRecsSearched = rQuery.totalRecsToBeSearched
+	rQuery.queryCount = &structs.QueryCount{
+		TotalCount: uint64(len(response.Hits.Hits)),
+		EarlyExit:  true,
+	}
+
+	rQuery.StateChan <- &QueryStateChanData{
+		StateName:       QUERY_UPDATE,
+		QueryUpdate:     &QueryUpdate{QUpdate: QUERY_UPDATE_LOCAL},
+		PercentComplete: 100,
+	}
+	return nil
+}
+
+func GetNewQueryPipelineResponse(qid uint64) *structs.PipeSearchResponseOuter {
```

Comment:
same comment as above, rename this func based on what name you give for the var above

Reply:
addressed in the latest commit.